src/main/common/time.c

```diff
@@ -120,14 +121,84 @@ static bool rtcIsDateTimeValid(dateTime_t *dateTime)
            (dateTime->millis <= 999);
 }
 
-static void dateTimeWithOffset(dateTime_t *dateTimeOffset, dateTime_t *dateTimeInitial, int16_t minutes)
+#if defined(RTC_AUTOMATIC_DST)
+static int lastSundayOfMonth(int currentYear, int wantedMonth) {
+    int days[] = {31,29,31,30,31,30,31,31,30,31,30,31};
+	int m, y = currentYear, w;
```

Comment:
This line seems very hairy. `m` should be declared in the `for`, since it's initialised and used only there. `y` is assigned to `w` (where is that variable coming from?), but it looks like you could just use `currentYear` a few lines down without any extra variables.